### ISSUE TITLE
int_convert: Memory effect regarding number-base if parser is used multiple times

### ISSUE THREAD

**Issue by jenisys** [r1chardj0n3s/parse#121]
RELATED TO NUMBER-FORMAT: `:d`
Bug was probably introduced in v1.15.0 when „int_convert()“ function was replaced by Callable class that remembers number-base auto-detection as attribute. When the parser object is used the second time, the first discovered number-base is used instead of performing auto-discovery of base again.

EXAMPLE:
```py
import parse
parser = parse.Parser(„{:d}“)
assert parser.parse(„0x12“).fixed[0] == 18.  # OK, but now base=16 was discovered in type-converter
assert parser.parse(„12“) .fixed[0] == 12  # FAILS: 18 again, because base=16 is used again instead of using base=10
```

**Comment by jenisys** [comment]
LESSON LEARNT: Type converters should be immutable (before first usage).

**Comment by jenisys** [comment]
DUPLICATES: #119

**Comment by r1chardj0n3s** [comment]
fixed in #119

**Comment by r1chardj0n3s** [comment]
Released as 0.18.0
